src/AerospikeCache.php

```diff
@@ -0,0 +1,130 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\AerospikeCache;
+
+use Symfony\Component\Cache\Adapter\AbstractAdapter;
+
+class AerospikeCache extends AbstractAdapter
+{
+    private const WRAPPER_NAME = 'data';
+
+    /** @var \Aerospike */
+    private $aerospike;
+
+    /** @var string */
+    private $namespace;
+
+    /** @var string */
+    private $set;
+
+    public function __construct(
+        \Aerospike $aerospike,
+        string $namespace,
+        string $set,
+        string $cacheNamespace = '',
+        int $defaultLifetime = 0
+    ) {
+        $this->aerospike = $aerospike;
+        $this->namespace = $namespace;
+        $this->set = $set;
+        parent::__construct($cacheNamespace, $defaultLifetime);
+    }
+
+    protected function doFetch(array $ids): array
+    {
+        $result = [];
+
+        $keys = $this->initializeKeysForAerospike($ids);
+
+        $this->aerospike->getMany($keys, $records);
+
+        foreach ($records as $record) {
+            if ($record['metadata'] !== null) {
+                $result[$record['key']['key']] = $record['bins'][self::WRAPPER_NAME] ?? null;
+            }
+        }
+
+        return $result;
+    }
+
+    private function initializeKeysForAerospike(array $ids): array
+    {
+        return array_map(
+            function ($id) {
+                return $this->createKey($id);
+            },
+            $ids
+        );
+    }
+
+    protected function doHave($id)
+    {
+        return $this->aerospike->get($this->createKey($id), $ignoredRecord) === \Aerospike::OK;
+    }
+
+    protected function doClear($namespace = ''): bool
+    {
+        if ($namespace === '') {
+            $statusCode = $this->aerospike->truncate($this->namespace, $this->set, 0);
+            $cleared = $this->isStatusOkOrNotFound($statusCode);
+        } else {
+            $removedAllRecords = true;
+            $clearNamespace = function ($record) use ($namespace, &$removedAllRecords): void {
+                if ($namespace === mb_substr($record['key']['key'], 0, mb_strlen($namespace))) {
```

Comment:
This could be used for better readability (as suggested by [PHP Inspections](https://plugins.jetbrains.com/plugin/7622-php-inspections-ea-extended-) plugin)

```suggestion
                if (mb_strpos($record['key']['key'], $namespace) === 0) {
```